examples/transports/README.md

```diff
@@ -96,78 +76,65 @@ That `QmW2cKTakTYqbQkUzBTEGXgWYFj1YEPeUndE1YWs6CBzDQ` is the PeerId that was cre
 
 ## 2. Dialing from one node to another node
 
-Now that we have our bundle, let's create two nodes and make them dial to each other! You can find the complete solution at [2.js](./2.js).
+Now that we have our `createNode` function, let's create two nodes and make them dial to each other! You can find the complete solution at [2.js](./2.js).
 
 For this step, we will need one more dependency.
 
 ```bash
-> npm install pull-stream
+> npm install it-pipe it-buffer
 ```
 
 And we also need to import the module on our .js file:
 
 ```js
-const pull = require('pull-stream')
+const pipe = require('it-pipe')
+const { toBuffer } = require('it-buffer')
 ```
 
-We are going to reuse the MyBundle class from step 1, but this time to make things simpler, we will create two functions, one to create nodes and another to print the addrs to avoid duplicating code.
+We are going to reuse the `createNode` function from step 1, but this time to make things simpler, we will create another function to print the addrs to avoid duplicating code.
 
 ```JavaScript
-function createNode (callback) {
-  let node
-
-  waterfall([
-    (cb) => PeerInfo.create(cb),
-    (peerInfo, cb) => {
-      peerInfo.multiaddrs.add('/ip4/0.0.0.0/tcp/0')
-      node = new MyBundle({ peerInfo: peerInfo })
-      node.start(cb)
-    }
-  ], (err) => callback(err, node))
-}
-
 function printAddrs (node, number) {
   console.log('node %s is listening on:', number)
   node.peerInfo.multiaddrs.forEach((ma) => console.log(ma.toString()))
 }
-
-```
-
-Now we are going to use `async/parallel` to create two nodes, print their addresses and dial from one node to the other. We already added `async` as a dependency, but still need to import `async/parallel`:
-
-```js
-const parallel = require('async/parallel')
 ```
 
 Then,
 
 ```js
-parallel([
-  (cb) => createNode(cb),
-  (cb) => createNode(cb)
-], (err, nodes) => {
-  if (err) { throw err }
-
-  const node1 = nodes[0]
-  const node2 = nodes[1]
+;(async () => {
+  const [peerInfo1, peerInfo2] = await Promise.all([
+    PeerInfo.create(),
+    PeerInfo.create()
+  ])
+  const [node1, node2] = await Promise.all([
+    createNode(),
+    createNode()
+  ])
 
   printAddrs(node1, '1')
   printAddrs(node2, '2')
 
-  node2.handle('/print', (protocol, conn) => {
-    pull(
-      conn,
-      pull.map((v) => v.toString()),
-      pull.log()
+  node2.handle('/print', ({ stream }) => {
+    pipe(
+      stream,
+      toBuffer,
+      source => (async function () {
+        for await (const msg of source) {
+          console.log(msg.toString())
```

Comment:
This will print the messages all on new lines so it looks like:
```sh
Hello
 
p2p
 
world
!
```

It might be better to collect everything and then do the console log. You can either use `toBuffer` and `concat` from streaming-iterables, or use [it-concat](https://github.com/alanshaw/it-concat) which will allow you to collect the BufferList and then just toString it.

```js
const concat = require('it-concat')
const result = await pipe(
  stream,
  concat
)
console.log(result.toString())
```